slsim/image_simulation.py

```diff
@@ -164,34 +185,44 @@ def rgb_image_from_image_list(image_list, stretch):
     return image_rgb
 
 
-def centered_coordinate_system(num_pix, transform_pix2angle):
+def centered_coordinate_system(num_pix, transform_pix2angle, angle_rot=90):
     """Returns dictionary for Coordinate Grid such that (0,0) is centered with given
     input orientation coordinate transformation matrix.
 
     :param num_pix: number of pixels
     :type num_pix: int
     :param transform_pix2angle: transformation matrix (2x2) of pixels into coordinate
         displacements
+    :param angle_rot: rotation angle of the coordinate system
     :return: dict with ra_at_xy_0, dec_at_xy_0, transfrom_pix2angle
     """
     pix_center = (num_pix - 1) / 2
-    ra_center = (
-        pix_center * transform_pix2angle[0, 0] + pix_center * transform_pix2angle[1, 0]
-    )
-    dec_center = (
-        pix_center * transform_pix2angle[0, 1] + pix_center * transform_pix2angle[1, 1]
-    )
-
+    rotation_angle = (
+        angle_rot * np.pi / 180
+    )  # rotation angle in North to West in radian
+    scale1_1 = -np.cos(rotation_angle)
```

Comment:
you have quite some duplicating code here to construct the rotation matrix. I'd suggest making an util function for it